src/services/completions.ts

```diff
@@ -3886,7 +3886,8 @@ function getCompletionData(
         // If already typing an import statement, provide completions for it.
         if (importStatementCompletion) return true;
         // If current completion is for non-contextual Object literal shortahands, ignore auto-import symbols
-        if (isNonContextualObjectLiteral) return false;
+        // if current non-contextual Object literal is spread assignment, we may need auto-import symbols

```

Comment:
1. The existing comment says to ignore auto-import symbols for non-contextual object literal *shorthands*. Why? A shorthand assignment references a variable the same way that a spread assignment does. Should auto-import work for `{ foo/**/ }` as well?
2. I think the existing comment is wrong since isNonContextualObjectLiteral doesn't appear to check for shorthand assignments.
3. The two comment lines should be combined, and if I'm right about (1) and (2), something like "If current completion is for non-contextual object literal, ignore auto-import symbols for property assignments, but not spread assignments or shorthand property assignments."

Reply:
1.  I found that auto-import does not work for `{ foo/**/ }`, I think it would be better if it works. But I haven't found a good way to detect this case, any suggestions?
2. Yes, I think these two comments should be combined.

Reply:
For `{ foo/**/ }`, currentToken.parent is considered an ObjectLiteralExpression node.

Reply:
Looking back at this, I don't understand why object literals without contextual types would ignore imports at all: they may not have a contextual type, but identifiers mentioned in them might still refer to potential imports.

For the case `{ foo/**/ }` where there is a contextual type, you can guess whether `foo` is a property name or not, and if it is, you don't want to auto-import unless the syntax is clearly a shorthand assignment. I think. But that's not what the code seems to implement. 

Non-contextually typed object literals don't have any information about expected property names -- maybe that's why they're not supposed to auto-import, because it's possible that any shorthand property assignment could turn into a normal property assignment?

@andrewbranch you've spent a lot of time thinking about this. Can you help me figure out what the right thing is?

Reply:
I think I agree there’s no strong reason to block an auto-import on a non-contextual object property. We could try removing it.